luarules/gadgets/cmd_nano_range_check.lua

```diff
@@ -0,0 +1,156 @@
+function gadget:GetInfo()
+	return {
+		name = "Construction Turrets Range Check",
+		desc = "Stops construction turrets from getting stuck on orders out of reach.",
+		author = "Nehroz",
+		date = "2024.12.01",
+		layer = 0,
+		enabled = true,
+	}
+end
+
+if not gadgetHandler:IsSyncedCode() then return end
+
+local CMD_STOP = CMD.STOP
+local gMaxUnits = Game.maxUnits
+local updateInterval = Game.gameSpeed / 2
+
+local spValidUnitID = Spring.ValidUnitID
+local spValidFeatureID = Spring.ValidFeatureID
+local spGetUnitDefID = Spring.GetUnitDefID
+local spGetUnitSeparation = Spring.GetUnitSeparation
+local spGetUnitFeatureSeparation = Spring.GetUnitFeatureSeparation
+local spGetUnitCommands = Spring.GetUnitCommands
+local spGiveOrderToUnit = Spring.GiveOrderToUnit
+
+--- Key is the nano's ID, followed by a set-like with the unitID of target currently inside the nano's range.
+--- Used for CommandArray manipulation
+---@type {[integer] : {integer : {cmdTag : integer, status : integer}}}
+local trackingTable = {}
+local tagUpdateFrameMax = 1 -- Can be bigger than 1 if you want game freezes.
+
+local statuses = {
+	outOfRange = 0,
+	inRange = 1,
+}
```

Comment:
Will there be other statuses in the future? This looks like a boolean isInRange.

Reply:
Depending on what players think of the current behaviour there could be more to add, e.x. a proximity check for close misses as a secondary range check.